snuba/reader.py

```diff
@@ -0,0 +1,98 @@
+from __future__ import annotations
+
+import re
+from abc import ABC, abstractmethod
+from datetime import datetime
+from typing import TYPE_CHECKING, Any, Mapping, Optional, Sequence
+
+from dateutil.tz import tz
+
+
+if TYPE_CHECKING:
+    from mypy_extensions import TypedDict
```

Comment:
These annotations aren't evaluated at runtime, and I didn't want to make `mypy` a hard dependency quite yet, so this is only evaluated when [`TYPE_CHECKING`](https://docs.python.org/3/library/typing.html#typing.TYPE_CHECKING).